docs/tutorials/cpp.md

```diff
@@ -1,43 +1,75 @@
 
-## Building and linking ICU4X
-
-- `cargo build` only supports local crates, so you have to download the published code from crates.io
-    - You can use `cargo clone icu_capi_staticlib@1.0.0` to do this (`cargo install cargo-clone` if the command is not installed)
-- Inside the crate you can use `cargo build` to build the library
-    - Be sure to pass `--release` to get an optimized build
-    - Specify Cargo features with `--features ...`:
-        - `compiled_data` \[default\] to include data (`ICU4XDataProvider::create_compiled()`)
-        - `simple_logger` \[default\] enable basic stdout logging of error metadata. Further loggers can be added on request.
-        - `buffer_provider` for working with blob data providers (`ICU4XDataProvider::create_from_byte_slice()`)
-        - `provider_fs` for loading data from the file system (`ICU4XDataProvider::create_fs()`). This also requires enabling a syntax on the `icu_provider` crate.
-    - Set `CARGO_PROFILE_RELEASE_LTO=true` to enable link-time optimization
-    - Set `CARGO_PROFILE_RELEASE_OPT_LEVEL="s"` to optimize for size
-    - See [cargo profiles](cargo-profiles) for more options
- - Copy the header files from the `icu_capi` crate
-    - If you're using the default crate registry, they will be in `~/.cargo/registry/src/*/icu_capi-1.0.0/cpp/include`
-    - If you cannot find the crate, download it from crates.io
-    - Make sure to use the same version as above
-        - While we selected a version for `icu_capi_staticlib` before, and here we're looking at `icu_capi`, those versions will always be in sync. You can sanity check this by running `cargo pkgid -p icu_capi`.
- - Compile with `g++ -std=c++17 icu/target/release/libicu_capi_staticlib.a -ldl -lpthread -lm`. C++ versions beyond C++17 are supported, as are other C++ compilers.
+## Building ICU4X
+
+After [installing Rust](https://rustup.rs), create a local Rust crate to configure ICU4X:
+
+```shell
+cargo init icu --lib
+cd icu
+echo "[lib]" >> Cargo.toml
+echo "crate-type = ['cdylib', 'staticlib']" >> Cargo.toml
+```
+
+Now, add the `icu_capi` crate with the desired features:
+
+```shell
+cargo add icu_capi --features foo
+echo "extern crate icu_capi;" > src/lib.rs
+```
+
+Features include:
+- `compiled_data` \[default\] to include data (`ICU4XDataProvider::create_compiled()`)
+- `simple_logger` \[default\] enable basic stdout logging of error metadata. Further loggers can be added on request.
+- `default_components` \[default\] activate all stable ICU4X components. For smaller builds, this can be disabled, and components can be added with features like `icu_list`.
+- `buffer_provider` for working with blob data providers (`ICU4XDataProvider::create_from_byte_slice()`)
+
+Default features can be turned off with
+```
+cargo add icu_capi --no-default-features
+```
+
+You can now build the library:
+
+```shell
+cargo build --release
+cd ..
```

Comment:
Issue: It's hard to keep track of directories in tutorials that move around. I normally like to prefix every box of shell commands with the working directory they are intended to be run from.